src/App.js

```diff
@@ -0,0 +1,254 @@
+        <li>Upload your CSV file.</li>
+        <li>You can view the results by clicking on the <b>Previous</b> and <b>Next</b> buttons.</li>
+        <li>To add a status or test result, click on <b>Accept</b> button to mark the data as passed. Otherwise, click on the <b>Reject</b> button.</li>
+        <li>Once you are done, download the CSV file.</li>
+      </ol>
+      
+    </div>
+  )
+}
+const Navigation = (props) => {
+  return (
+    <div className="navigation">
+      <button onClick={props.onPrev}>{'\u2B05'} Previous</button>
+      <button onClick={props.onModalClose}>Close</button>
+      <button onClick={props.onNext}>Next {'\u27A1'}</button>    
+    </div>
+  )
+}
+
+const Download = (props) => {
+  return (
+    <div className="download">
+      <button id="downloadBtn" onClick={props.onDownload} value="download">Download</button>
+    </div>
+  )
+}
+
+const Preview = (props) => {
+  return (
+    <div id="preview-all">
+      <div className="column">
+        <p>
+          <strong>PAGE {props.activeData + 1}:</strong>
+          <a href={props.oldUrl}> {props.oldUrl}</a>
+        </p>
+        <iframe title="preview-0" className="preview" src={'https://image.thum.io/get/width/1024/fullpage/allowJPG' + props.oldUrl} width="100%" height="500" />
+      </div>
+      <div className="column">
+        <p>
+          <strong>PAGE {props.activeData + 1}:</strong>
+          <a href={props.newUrl}> {props.newUrl}</a>
+        </p>
+        <iframe title="preview-1" className="preview" src={'https://image.thum.io/get/width/1024/fullpage/allowJPG' + props.newUrl} width="100%" height="500" />
+      </div>
+    </div>
+  )
+}
+
+const Modal = (props) => {
+  return (
+      <div>
+        <div id="status">
+          <span id="status-text"></span>
+          <p>
+            <strong>Status:</strong> {props.status}
+          </p>
+        </div>
+        <div className="action">
+          <button className="accept" onClick={props.onAccept}>Accept</button>
+          <button className="reject" onClick={props.onReject}>Reject</button>
+        </div>
+        <Preview {...props}></Preview>
+        <div className="skip">
+          <p>To skip to the item without status: </p>
+          <button className="skip-button" onClick={props.onSkip}>Skip</button>
+        </div>
+        <Download 
+          onDownload={props.onDownload}>
+        </Download>
+        <Navigation {...props}></Navigation>
+      </div>
+  );
+}
+
+function App() {
+  const [parsedData, setParsedData] = useState([]);
+  const [newData, setNewData] = useState([]);
+  const [activeData, setActiveData] = useState('');
+  const [selectedFile, setSelectedFile] = useState();
+	const [isFilePicked, setIsFilePicked] = useState(false);
+  const [oldUrl, setOldUrl] = useState('');
+  const [newUrl, setNewUrl] = useState('');
+  const [status, setStatus] = useState('');
+
```

Comment:
**Suggestion (non-blocking)**: Another tip that I've found helpful is to minimize what you are tracking as state, and to try to derive or calculate other values from that set of state. State should only be confined to what root data needs to be directly manipulated. This can reduce a lot of complexity by having a single "source of truth."

So, as a couple of examples...

- `oldUrl` and `newUrl` could just be calculated from `parsedData` (or `newData`) and the `activeData` index.
- `status` probably doesn't need to be its own piece of state and can just be derived and set on the `newData` state
- `isFilePicked` could be calculated from `selectedFile` not being undefined

Reply:
> Another tip that I've found helpful is to minimize what you are tracking as state, and to try to derive or calculate other values from that set of state.

This is so helpful!!! When I was working on these states, it was confusing for me because there were too many states that I need to "remember". Extracting the data directly from `newData` and `activeData` index simplified things. 

Thank you so much!